code/controllers/subsystem/movement/movement_types.dm

```diff
@@ -379,8 +381,18 @@
 	var/list/movement_path
 	///Cooldown for repathing, prevents spam
 	COOLDOWN_DECLARE(repath_cooldown)
+	///Bool used to determine if we're already making a path in JPS. this prevents us from re-pathing while we're already busy.
+	var/is_pathing = FALSE
+	///The initial path we received from an outside source. Can be used to save CPU if you had already calculated a path initially outside of the moveloop.
+	var/list/initial_path
```

Comment:
why do we hold a reference to this? why not just set movement_path to it on New?
It breaks if the loop is delayed but so does this implementation